src/lib.rs

```diff
@@ -357,13 +384,16 @@ impl AutoCfg {
     /// Any non-identifier characters in the `path` will be replaced with
     /// `_` in the generated config value.
     pub fn emit_has_path(&self, path: &str) {
+        let cfg_flag = format!("has_{}", mangle(path));
+        emit_possibility(&cfg_flag);
         if self.probe_path(path) {
-            emit(&format!("has_{}", mangle(path)));
+            emit(&cfg_flag);
         }
```

Comment:
I think we can now simplify this to `self.emit_path_cfg(path, &format!(...));` -- It was being lazy about the formatting before, but now we can just forward the call. Same for the other `emit_has_` methods too.